srml/consensus/src/lib.rs

```diff
@@ -118,7 +118,23 @@ decl_storage! {
 	trait Store for Module<T: Trait> as Consensus {
 		// Authorities set actual at the block execution start. IsSome only if
 		// the set has been changed.
-		OriginalAuthorities: Vec<T::SessionKey>;
+		OriginalAuthorities: Option<Vec<T::SessionKey>>;
+	}
+	add_extra_genesis {
+		config(authorities): Vec<T::SessionKey>;
+		#[serde(with = "substrate_primitives::bytes")]
+		config(code): Vec<u8>;
+
+		build(|storage: &mut primitives::StorageMap, config: &GenesisConfig<T>| {
+			use codec::{Encode, KeyedVec};
+
+			let auth_count = config.authorities.len() as u32;
+			for (i, v) in config.authorities.clone().into_iter().enumerate() {
```

Comment:
Instead of `for` you could just use the `for_each` combinator.

Reply:
Thanks, converted to `for_each`.